### DISCUSSION TITLE
How to set (change) wal_segment_size?

### DISCUSSION THREAD

**Discussion by loxon-git** [autobase-tech/autobase#775] [Q&A]
How to set (change) wal_segment_size before the cluster installation (and after if needed)?

**Comment by vitabaks** [comment]
Hello.

a different value for the cluster may have been set by with the [initdb](https://www.postgresql.org/docs/16/app-initdb.html#APP-INITDB-OPTION-WAL-SEGSIZE) option `--wal-segsize`

Try adding the option to the [patroni.yml.j2](https://github.com/vitabaks/postgresql_cluster/blob/2.0.0/automation/roles/patroni/templates/patroni.yml.j2#L123) template and deploy new cluster.

**Reply by loxon-git** [reply]
Hello,

Thank you, for the feedback. Finally, I have added a var "postgresql_wal_segsize" to vars/main.yml and the below code the template (works fine... at least during Installation).
Regards

  {% if postgresql_wal_segsize is defined %}
    - wal-segsize={{ postgresql_wal_segsize }}
  {% endif %}

**Reply by vitabaks** [reply]
Feel free to do a PR
